### PR TITLE
Add events module and EventEmitter

### PR DESCRIPTION
**Pull request by Mehuge**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/cu-patcher-ui/pull/1" target="_blank">ReviewNinja

File: src/ts/core/events.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+// Export interface
+
+export {
+  on,
+  once,
+  off,
+  fire
+};
```
Comment (codecorsair) [inline src/ts/core/events.ts]: new line at end of file is missing.

File: src/ts/core/EventEmitter.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    const listener: Listener = listeners[i];
+                    if (listener.once) {
+                        listeners[i] = null;
+                    }
+                    listener.callback(data);
+                }
+            }
+        }
+    }
+}
```
Comment (codecorsair) [inline src/ts/core/EventEmitter.ts]: new line at end of file is missing.

File: src/ts/core/events.ts (hunk 1 above)
Comment (Mehuge) [inline src/ts/core/events.ts]: !fixed

File: src/ts/core/EventEmitter.ts (hunk 2 above)
Comment (Mehuge) [inline src/ts/core/EventEmitter.ts]: !fixed

**Pull request merged**